### DISCUSSION TITLE
Should enum be hoistable declaration?

### DISCUSSION THREAD

**Discussion by Jack-Works** [Jack-Works/proposal-enum#18] [General]
Now the EnumDeclaration is a Declaration (like function, class, and const/let).

Should we make it hoistable like function?

## Problem

This conflicts with fully computed key (#11), fully computed value (#12), and extend other enums (#5) because it might trigger userland code and have observable side effects.

**Comment by rbuckton** [comment]
No, they should not be hoisted. I would like to be able to eventually define decorators on enums and enum members and would rather not have the same surprising semantics as decorators might have on function declarations. 

Hoisting would also be problematic if you initialized an enum member to the result of an expression. I'd much rather we follow the `class` semantics for this one.

**Comment by ljharb** [comment]
We shouldn’t add new hoisted things to the language.

**Comment by Jack-Works** [comment]
Ok. I'll add hoistable declaration as a non-starter.
